lib/lsp-data/api_pol_invoice_list.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module LspData
+  ### Represents the list of invoices returned from an invoice API call that
+  ###   finds invoices attached to a specific PO Line
+  class ApiPolInvoiceList
+    attr_reader :pol, :api_key, :conn
+
+    def initialize(pol:, api_key:, conn:)
+      @pol = pol
+      @api_key = api_key
+      @conn = conn
+    end
+
+    def invoices
+      @invoices ||= all_invoices
+    end
+
+    private
+
+    def all_invoices
+      initial_response = api_call(offset: 0)
+      total_invoice_count = initial_response[:body]['total_record_count']
+      results = initial_response[:body]['invoice'].map do |invoice|
+        ApiInvoice.new(invoice_json: invoice)
+      end
+      if total_invoice_count.size.positive?
+        total_calls = (total_invoice_count / 100).floor
+        1.upto(total_calls).each do |call|
+          info = api_call(offset: (call * 100))
+          results += info[:body]['invoice'].map do |invoice|
+            ApiInvoice.new(invoice_json: invoice)
+          end
+        end
+      end
+      @all_invoices = results
```

Comment:
```suggestion
      results
```

I'm not sure it's necessary to set the `@all_invoices` instance variable, especially since this is already memoized with the `@invoices` variable.